web-module/src/main/webapp/resources/structEditJS/StructEditor.mjs

```diff
@@ -0,0 +1,879 @@
+
+  _startGetFieldValueAsync : function(formfield) {
+    const _me = this;
+    if ((typeof tinyMCE !== 'undefined') && formfield.classList.contains('mceEditor')) {
+      if (tinyMCE.get(formfield.id)) {
+        _me._updateOneTinyMCETextArea(formfield);
+      } else {
+        console.debug('_startGetFieldValueAsync: still loading, adding delayed read ',
+          formfield.id);
+        return new Promise((resolve, reject) => {
+          console.debug('_startGetFieldValueAsync: start Promise ', formfield.id);
+          const finishLoadingTinymMce = function(event) {
+            const editor = event.memo.editor;
+            if (editor.id === formfield.id) {
+              _me._updateOneTinyMCETextArea(formfield, resolve, reject);
+              console.log('finishLoadingTinymMce: finished loading, updated tinyMCE for ',
+                editor.id);
+            } else {
+              console.debug('finishLoadingTinymMce: skip event for ', editor.id, ' waiting for ',
+                formfield.id);
+            }
+          };
+          $$('body')[0].observe('celRTE:finishedInit', finishLoadingTinymMce);
+        });
+      }
+    } else {
+      console.debug('_startGetFieldValueAsync: no tinyMCE or no mceEditor field ', formfield.id);
+    }
+    return null;
+  },
+
+  _getFieldValue : function(formfield, cblFunc) {
+    const _me = this;
+    console.debug('_startGetFieldValueAsync: _startGetFieldValueAsync for ', formfield.id);
+    const possiblePromise = _me._startGetFieldValueAsync(formfield, cblFunc);
+    if (possiblePromise) {
+      possiblePromise
+      .then(cblFunc)
+      .catch(exp => {
+        console.error('_getFieldValue: failed with exception ' + formfield.id, exp);
+      });
+    } else {
+      cblFunc(formfield.value);
+    }
+  },
+
+  _updateTinyMCETextAreas : function() {
+    const _me = this;
+    const formId = _me.getFormId();
+    const mceFields = document.forms[formId].select('textarea.mceEditor');
+    if (typeof tinyMCE !== 'undefined') {
+      console.log('_updateTinyMCETextAreas: for ', formId, mceFields);
+      mceFields.each(function(formfield) {
+        _me._updateOneTinyMCETextArea(formfield);
+      });
+    } else {
+      console.debug('_updateTinyMCETextAreas: skip no tinyMCE');
+    }
+    console.log('_updateTinyMCETextAreas: end ', formId);
+  },
+
+  /**
+   * submittable fields must have a name attribute and maynot be disabled
+   * @param fieldElem
+   * @returns {Boolean}
+   */
+  _isSubmittableField : function(fieldElem) {
+    return (fieldElem.name && (fieldElem.name != '') && !fieldElem.disabled);
+  },
+
+  _retrieveInitialValues : function() {
+    const formId = this.getFormId();
+    if (this.isValidFormId()) {
+      const formdata = new FormData(document.getElementById(formId));
+      const newInitialValues = {};
+      [...formdata.keys()].forEach(key => {
+		    console.log(key, formdata.getAll(key));
+		    newInitialValues[key] = formdata.getAll(key);
+		  });
```

Comment:
I don't think that we normally have to store more then 10 values for one key. Still if it would be 100 the performance impact is not messurable. Nevertheless I will add this camplexity if you request it.